code/game/machinery/computer/camera.dm

```diff
@@ -233,6 +232,16 @@
 	else
 		set_light(1, LIGHTING_MINIMUM_POWER)
 
+/obj/machinery/computer/security/telescreen/entertainment/wrench_act(mob/living/user, obj/item/I)
+	. = TRUE
+	if(!I.use_tool(src, user, 0 SECONDS))
+		return
+	TOOL_ATTEMPT_DISMANTLE_MESSAGE
+	if(I.use_tool(src, user, 2 SECONDS, volume = I.tool_volume))
+		TOOL_DISMANTLE_SUCCESS_MESSAGE
+		new /obj/item/mounted/frame/newscaster_frame/entertainment_frame(drop_location(), 1)
```

Comment:
Why not put this in the `deconstruct` proc?

Reply:
I think this proc is leading to the standard proc for machine deconstruction and i dont think it can be included there, what should i do in that case?

Reply:
Sorry to get back this late to you.
You can also put the creation of these in either `deconstruct` or `on_deconstruct`.  Goes for the other cases as well.
`deconstruct` is the proc itself which causes a machine to be deconstructed. `on_deconstruct` is called when it when `deconstruct` is also called. Either will work just fine as long as you call the parent in both

Reply:
Done